contracts/dnsregistrar/DNSRegistrar.sol

```diff
@@ -98,11 +85,39 @@ contract DNSRegistrar {
         claim(name, proof);
     }
 
+    function proveAndClaimWithResolver(bytes memory name, DNSSEC.RRSetWithSignature[] memory input, bytes memory proof, address resolver, address addr) public {
+        proof = oracle.submitRRSets(input, proof);
+        (bytes32 rootNode, bytes32 labelHash, address owner) = _claim(name, proof);
+        require(msg.sender == owner, "Only owner can call proveAndClaimWithResolver");
+        ens.setSubnodeRecord(rootNode, labelHash, owner, resolver, 0);
+        if(addr != address(0)) {
+            bytes32 node = keccak256(abi.encodePacked(rootNode, labelHash));
+            AddrResolver(resolver).setAddr(node, addr);
```

Comment:
Assuming the reverse registrar change happens at the same time, is it worth setting a reverse record as well?

Reply:
I'd rather just get this change in than further complicate it.